src/ports/Jwt.ts

```diff
@@ -0,0 +1,101 @@
+import cookie from 'cookie'
+import { CrossFetchRequest, Fetcher } from 'dcl-catalyst-commons'
+import log4js from 'log4js'
+import { generateNonceForChallenge } from '../utils/ProofOfWork'
+
+const LOGGER = log4js.getLogger('JWTPort')
+
+export async function obtainJWT(fetcher: Fetcher, catalystUrl: string): Promise<string | undefined> {
+  try {
+    const response = await fetcher.fetchJson(catalystUrl + '/pow-auth/challenge')
+    const body = JSON.parse(JSON.stringify(response))
+
+    const challenge: string = body.challenge
+    const complexity: number = body.complexity
+    const nonce: string = await generateNonceForChallenge(challenge, complexity)
+
+    const powAuthUrl = new URL('/pow-auth/challenge', catalystUrl).href
+    const challengeBody = JSON.stringify({ challenge: challenge, complexity: complexity, nonce: nonce })
+    const jwtResponse = await fetcher.postForm(powAuthUrl, { body: challengeBody })
+
+    if (!jwtResponse.jwt) {
+      LOGGER.warn('[POW] Could not get a JWT from Pow Auth Server.')
+    }
+    return jwtResponse.jwt
+  } catch (error) {
+    LOGGER.warn(`[POW] Could not get a JWT from Pow Auth Server, due to: ${error}`)
+    return ''
+  }
+}
+
+export function removedJWTCookie(response: Response): boolean {
+  try {
+    const headers = response.headers
+    if (headers) {
+      const setCookie = headers.get('Set-Cookie')
+      if (setCookie && setCookie.includes('JWT=')) {
+        const cookies = cookie.parse(setCookie)
+        return cookies.JWT == ''
+      }
+    }
+    return false
+  } catch {
+    return false
+  }
+}
+
+export function noJWTinCookie(request: CrossFetchRequest): boolean {
```

Comment:
We have too many functions that check if the cookie is present haha. Could we make their names clearer so that it is easier to understand the differences?